### PR TITLE
Support rank gauss normalization in feature transform.

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*

*Description of changes:*
Implement Rank Gauss. It maps the numeric feature values to gaussian distribution based on ranking. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/gconstruct/transform.py [hunk 1]
```diff
@@ -68,6 +70,25 @@ def feat_name(self):
         """
         return self._feat_name
 
+class GlobalProcessFeatTransform(FeatTransform):
+    """ The base class for transformations taht can only be done using a single process.
```
Comment (zheng-da) [inline python/graphstorm/gconstruct/transform.py:74]: taht => that

File: python/graphstorm/gconstruct/transform.py [hunk 2]
```diff
@@ -207,6 +228,53 @@ def __call__(self, feats):
...
+class RankGaussTransform(GlobalProcessFeatTransform):
+    """ Use Gauss rank transformation to transform input data
+
+        The idea is from
+        http://fastml.com/preparing-continuous-features-for-neural-networks-with-rankgauss/
+
+    Parameters
+    ----------
+    Parameters
+    ----------
```
Comment (zheng-da) [inline python/graphstorm/gconstruct/transform.py:240]: remove one of them

**Comment by zheng-da** [comment]
you should add tests in tests/unit-tests/gconstruct/test_construct_graph.py as well

**Comment by classicsong** [comment]
> you should add tests in tests/unit-tests/gconstruct/test_construct_graph.py as well

Added

**Review by zheng-da — APPROVED** [review]

**Pull request merged**